src/transformers/models/auto/modeling_auto.py

```diff
@@ -600,6 +604,16 @@
     ]
 )
 
+MODEL_FOR_DOCUMENT_QUESTION_ANSWERING_MAPPING_NAMES = OrderedDict(
+    [
+        # TODO Uncomment after # 18407 lands
+        # ("layoutlm", "LayoutLMForQuestionAnswering"),
+        ("donut-swin", "DonutSwinModel"),
```

Comment:
I don't think Donut-swin should be here, as this is against the philosophy. To clarify, any XXX_MAPPING_NAMES defines the models which are supported by a certain AutoModelForXXX. 

Here, LayoutLM, LayoutLMv2 and LayoutLMv3 could fit in this Auto class. But Donut not, as this one already fits into the MODEL_FOR_VISION_2_SEQ_MAPPING_NAMES.

Reply:
Fixed! Thanks for the explanation. @Narsil given that we're adding `LayoutLMForQuestionAnswering` here, should we just avoid adding it to `AutoModelForQuestionAnswering` in PR #18407?

Reply:
The AutoModel implementations, should only be done regarding what the model actually implements and which heads it actually support, it shouldn't be linked to what the pipeline does.

Reply:
Got it, thank you. The current code _only_ adds `("layoutlm", "LayoutLMForQuestionAnswering")` to `MODEL_FOR_DOCUMENT_QUESTION_ANSWERING_MAPPING_NAMES`. I will tag you guys there to pick up the discussion.